diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/StringConcatenationRule.kt

```diff
@@ -155,7 +156,8 @@ class StringConcatenationRule(configRules: List<RulesConfig>) : DiktatRule(
             val rvalueTextNew = rvalueText?.trim('"')
             return "$lvalueText$rvalueTextNew"
         } else if (binaryExpressionPsi.isRvalueCallExpression() || binaryExpressionPsi.isRvalueDotQualifiedExpression() ||
-                binaryExpressionPsi.isRvalueOperation() || binaryExpressionPsi.isRvalueSafeQualifiedExpression()) {
```

Comment:
Do we need this branch at all? Or can it be the default behaviour (like the last branch of `if`) when right side is a general KtExpression? I don't like that we need to add so many types here, I bet it will fire once again on something else.

Reply:
Yes, I agree.
Fixed

Reply:
What is the reason to leave separate branch for call expression?

Reply:
ReferenceExpression is always a child for the CallExpression. That's why the CallExpression requires a separate condition.
example
- foo()  [CallExpression]
- - foo [ReferenceExpression]

Reply:
But then `isRvalueReferenceExpression()` will be false and we'll get in `.isRvalueExpression()` condition, won't we?

Reply:
No isRvalueReferenceExpression return true if parent or one of the children will be True